src/irmin/tree.ml

```diff
@@ -270,6 +269,8 @@ module Make (P : Private.S) = struct
           | Some x, Some y -> equal_contents x y
           | _ -> equal_hash (hash x) (hash y))
 
+    let t = Type.map ~equal:(Type.stage equal) v of_v (fun t -> t.v)
```

Comment:
That seems like a good fix (as it's probably faster to compare cached hashes than the nodes) but I'm wondering why is this part of that PR.

Reply:
Because content values are no longer forced from hashes when importing, this PR would otherwise break tests that rely on being able to round-trip to `concrete` trees w/o breaking equality. I can fix the tests to force in the appropriate places, but decided it's probably better to just have a better notion of equality here.

Reply:
sounds good